modules/gapi/include/opencv2/gapi/infer/ie.hpp

```diff
@@ -117,75 +140,180 @@ template<typename Net> class Params {
+    */
     Params& cfgNumRequests(size_t nireq) {
         GAPI_Assert(nireq > 0 && "Number of infer requests must be greater than zero!");
         desc.nireq = nireq;
         return *this;
     }
 
-    Params<Net>& cfgInputReshape(std::map<std::string, std::vector<std::size_t>>&& reshape_table) {
-        desc.reshape_table = std::move(reshape_table);
-        return *this;
-    }
+    /** @brief Specifies new input shapes for the network inputs.
 
+    The function is used to set new input shapes for network. You can specify
+    new dimensions for one or some or all of network input layers. network will be
+    reshaped based on this information and your inputs will be preprocessed for
+    new input sizes. Follow https://docs.openvinotoolkit.org/latest/classInferenceEngine_1_1networkNetwork.html
+    for additional information.
+
+    @param reshape_table Map of pairs: name of corresponding data and its dimension.
+    @return the reference on modified object.
+    */
     Params<Net>& cfgInputReshape(const std::map<std::string, std::vector<std::size_t>>& reshape_table) {
         desc.reshape_table = reshape_table;
         return *this;
     }
 
-    Params<Net>& cfgInputReshape(std::string&& layer_name, std::vector<size_t>&& layer_dims) {
-        desc.reshape_table.emplace(layer_name, layer_dims);
+    /** @overload
+    Function with a rvalue parameters.
+
+    @param reshape_table rvalue map of pairs: name of corresponding data and its
+    dimension.
+    @return the reference on modified object.
+    */
+    Params<Net>& cfgInputReshape(std::map<std::string, std::vector<std::size_t>>&& reshape_table) {
+        desc.reshape_table = std::move(reshape_table);
         return *this;
     }
 
+    /** @overload
+    Function for one input layer.
+
+    @param layer_name Name of layer.
+    @param layer_dims New dimensions for this layer.
+    @return the reference on modified object.
+    */
     Params<Net>& cfgInputReshape(const std::string& layer_name, const std::vector<size_t>& layer_dims) {
         desc.reshape_table.emplace(layer_name, layer_dims);
         return *this;
     }
 
-    Params<Net>& cfgInputReshape(std::unordered_set<std::string>&& layer_names) {
-        desc.layer_names_to_reshape = std::move(layer_names);
+    /** @overload
+    Function with a rvalue parameters for one input layer.
+
+    @param layer_name rvalue name of layer.
+    @param layer_dims rvalue new dimensions for this layer.
+    @return the reference on modified object.
+    */
+    Params<Net>& cfgInputReshape(std::string&& layer_name, std::vector<size_t>&& layer_dims) {
+        desc.reshape_table.emplace(layer_name, layer_dims);
         return *this;
     }
 
+    /** @overload
+    Function for reshape by image size.
+
+    The function is used to set names of layers that will be used for network reshape.
+    Dimensions will be constructed automatically by current network input and height and
+    width of image.
```

Comment:
Please align with the rest of the reshape documentation

Reply:
@TolyaTalamanov ?

Reply:
Done